pycalrissian/job.py

```diff
@@ -249,6 +249,39 @@ def to_k8s_job(self):
 
             volume_mounts.append(pod_node_selector_volume_mount)
 
+        try:
+            workspace_config = self.runtime_context.core_v1_api.read_namespaced_config_map(name="workspace-config", namespace=self.runtime_context.namespace)
+        except Exception as e:
+            logger.error(f"Failed to read 'workspace-config' ConfigMap: {e}")
+            workspace_config = None
+        pvcs_json = workspace_config.data.get("pvcs", "[]")
+        try:
+            pvcs_list = json.loads(pvcs_json)
+        except json.JSONDecodeError as e:
+            logger.error(f"Error parsing PVCs JSON: {e}")
+            pvcs_list = []
+
+        for pvc_map in pvcs_list:
+            pvc_name = pvc_map.get("pvcName")
+            if pvc_name and self.runtime_context.is_pvc_created(name=pvc_name):
+                volume_name = f"workspace-efs-{pvc_name}"
+                efs_pvc_volume = client.V1Volume(
+                    name=volume_name,
+                    persistent_volume_claim=client.V1PersistentVolumeClaimVolumeSource(
+                        claim_name=pvc_name
+                    ),
+                )
+
+                efs_volume_mount = client.V1VolumeMount(
+                    mount_path=pvc_map.get("rootDirectory"),
```

Comment:
Expect an update here after discussion around the expected mount path being `/workspace/<block-store-name>`. Once the config map is updated to include the PVName this should just construct the expected mount path based on that value.
See review comment [here](https://github.com/UKEODHP/workspace-controller/pull/11/files#r1682839545)

Reply:
Updated as discussed